### DISCUSSION TITLE
[@types/ibm_db] Database's close with callback and close as promise

### DISCUSSION THREAD

**Discussion by Stoyz** [DefinitelyTyped/DefinitelyTyped#60620] [Issues with a @types package]
I think there is a mistake in setting the `close(cb?: (err: Error, db: Database) => any): void;` ([line 66](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/ibm_db/index.d.ts#L66=)) method before the `close(): Promise<void>;` method ([line 67](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/ibm_db/index.d.ts#L67=)) because typescript informs me that "'await' has no effect on the type of this expression" when trying to await che close method without parameters.

In my local `index.d.ts` file I have tried switching lines position and it worked without warnings.

I am quite new in typescript so if I am doing something wrong I would appreciate any suggestion.

[a comment by an automated account is not shown]

**Comment by Grayson-code** [comment]
fixed in #60870
